src/lib/homestore_backend/hs_pg_manager.cpp

```diff
@@ -0,0 +1,157 @@
+    return folly::makeSemiFuture< PGManager::NullResult >(folly::makeUnexpected(PGError::UNSUPPORTED_OP));
+}
+
+void HSHomeObject::add_pg_to_map(shared< HS_PG > hs_pg) {
+    RELEASE_ASSERT(hs_pg->pg_info_.replica_set_uuid == hs_pg->repl_dev_->group_id(),
+                   "PGInfo replica set uuid mismatch with ReplDev instance for {}",
+                   boost::uuids::to_string(hs_pg->pg_info_.replica_set_uuid));
+    auto lg = std::scoped_lock(_pg_lock);
+    auto id = hs_pg->pg_info_.id;
+    auto [it1, _] = _pg_map.try_emplace(id, std::move(hs_pg));
+    RELEASE_ASSERT(_pg_map.end() != it1, "Unknown map insert error!");
+}
+
+#if 0
+std::string HSHomeObject::serialize_pg_info(PGInfo const& pginfo) {
+    nlohmann::json j;
+    j["pg_info"]["pg_id_t"] = pginfo.id;
+    j["pg_info"]["repl_uuid"] = boost::uuids::to_string(pginfo.replica_set_uuid);
+
+    nlohmann::json members_j = {};
+    for (auto const& member : pginfo.members) {
+        nlohmann::json member_j;
+        member_j["member_id"] = member.id;
+        member_j["name"] = member.name;
+        member_j["priority"] = member.priority;
+        members_j.push_back(member_j);
+    }
+    j["pg_info"]["members"] = members_j;
+    return j.dump();
+}
+
+PGInfo HSHomeObject::deserialize_pg_info(std::string const& json_str) {
+    auto pg_json = nlohmann::json::parse(json_str);
+
+    PGInfo pg_info;
+    pg_info.id = pg_json["pg_info"]["pg_id_t"].get< pg_id_t >();
+    pg_info.replica_set_uuid = boost::uuids::string_generator()(pg_json["pg_info"]["repl_uuid"].get< std::string >());
+
+    for (auto const& m : pg_info["pg_info"]["members"]) {
+        PGMember member;
+        member.id = m["member_id"].get< pg_id_t >();
+        member.name = m["name"].get< std::string >();
+        member.priority = m["priority"].get< int32_t >();
+        pg_info.members.emplace(std::move(member));
+    }
+    return pg_info;
+}
+#endif
+
+void HSHomeObject::on_pg_meta_blk_found(sisl::byte_view const& buf, void* meta_cookie) {
+    homestore::superblk< pg_info_superblk > pg_sb;
+    pg_sb.load(buf, meta_cookie);
+
+    hs_repl_service()
+        .open_repl_dev(pg_sb->replica_set_uuid, std::make_unique< ReplicationStateMachine >(this))
+        .thenValue([this, pg_sb = std::move(pg_sb)](auto&& v) {
+            if (v.hasError()) {
+                // TODO: We need to raise an alert here, since without pg repl_dev all operations on that pg will fail
+                LOGERROR("open_repl_dev for group_id={} has failed", boost::uuids::to_string(pg_sb->replica_set_uuid));
+                return;
+            }
+            add_pg_to_map(std::make_shared< HS_PG >(pg_sb, std::move(v.value())));
+        });
+}
+
+PGInfo HSHomeObject::HS_PG::pg_info_from_sb(homestore::superblk< pg_info_superblk > const& sb) {
+    PGInfo pginfo{sb->id};
+    for (uint32_t i{0}; i < sb->num_members; ++i) {
+        pginfo.members.emplace(sb->members[i].id, std::string(sb->members[i].name), sb->members[i].priority);
+    }
```

Comment:
pginfo.replica_set_uuid = sb->replica_set_uuid; 
it is needed in my local env test when HO restarts,  otherwise L66 assert() will be failed.
it is a smaller change, it can be fixed in this pr or use another pr.